x-pack/plugins/spaces/public/management/edit_space/roles/component/space_assigned_roles_table.tsx

```diff
@@ -140,33 +142,43 @@ const getTableColumns = ({
       ),
       actions: [
         {
-          type: 'icon',
-          icon: 'lock',
-          href: '#',
-          target: '_self',
-          'data-test-subj': 'spaceRoleCellActionLocked',
-          name: (role) =>
-            isRoleReserved(role)
-              ? i18n.translate(
-                  'xpack.spaces.management.spaceDetails.rolesTable.column.actions.notEditableTitle.isReserved',
-                  { defaultMessage: 'Reserved' }
-                )
-              : i18n.translate(
-                  'xpack.spaces.management.spaceDetails.rolesTable.column.actions.notEditableTitle.isAssignedToAll',
-                  { defaultMessage: 'Assigned to all spaces' }
-                ),
-          description: (role) =>
-            isRoleReserved(role)
-              ? i18n.translate(
-                  'xpack.spaces.management.spaceDetails.rolesTable.column.actions.notEditableDescription.isReserved',
-                  { defaultMessage: `You can’t edit the access of reserved roles to this space.` }
-                )
-              : i18n.translate(
-                  'xpack.spaces.management.spaceDetails.rolesTable.column.actions.notEditableDescription.isAssignedToAll',
-                  {
-                    defaultMessage: `You can't edit the access of a role that is assigned to all spaces.`,
+          render: (role) => {
```

Comment:
This changes the action to use a [custom action](https://github.com/elastic/eui/blob/main/packages/eui/src/components/basic_table/action_types.ts#L88)